### ISSUE TITLE
Synth shell is a bit slow in Windows WSL

### ISSUE THREAD

**Issue by vinrepos** [andresgongora/synth-shell-prompt#35]
Hi so I've managed to make synth shell look exactly how I wanted but I just found it to be a bit slow. Sometimes it takes like half-a-second to load the output. Is there any way to speed it up? I've also attached a video recording of my terminal. 

https://github.com/andresgongora/synth-shell-prompt/assets/104124987/034069f3-d263-49ae-b5e0-20403f9a9aed

**Comment by andresgongora** [comment]
Usually the slow bit is because it tries to get its public IP but cant
reach anyone to get this info. Can you confirm that the slow response only
happens if you have no internet? In either case, you can go to config and
remove the PUBLIC_IPV4 (i think it was called something like that) and see
if it helps. Please report back and we'll take it from there :)

**Comment by vinrepos** [comment]
Hey I do have internet connectivity on my host machine and the wsl. 

And I couldn't find any IP address related lines in the synth shell config files anywhere. Could you let me know where exactly it would be, please?

Thanks
vin

**Comment by andresgongora** [comment]
Sure! Under .config/synth-shell/synth-shell-greeter.config (the exact path
to the config path is probably is probably a bit weird for WSL, I don't use
it, so I can't tell). Once you are in the config file, search for the line
`print_info="`. Under it you will see a list of all the elements that get
printed when you open a terminal. One of these elements is `EXTERNALIPV4`.
Simply remove the line that contains it and try to open a terminal. You'll
see that there is now one element less in the greeting message; you will
have removed your external IPv4 from the list. I'd bet this is the element
that makes your machine slightly slower since it's the only one that has to
"communicate" to the outside world (it gets the IP from Google, I believe,
we wrote this a looong time ago). Let me know if it worked :)

**Comment by vinrepos** [comment]
Hi  thanks, but I don't have a synth-shell-greeter.config file. Have a look at the screenshot below, please.

![image](https://github.com/andresgongora/synth-shell-prompt/assets/104124987/9b49eb1a-25dc-41dd-825d-ead131bf60bb)

**Comment by andresgongora** [comment]
Ahh I see, you only installed the prompt!! My bad. I'm used to people
installing the whole synt-shell package.
In that case... I fear I dont know what could be introducing that extra
delay.
When you disable synth-shell-prompt from your terminal, does the lag go
away?

**Comment by vinrepos** [comment]
Yeah I'm using only the synth shell prompt. I just tried uninstalling it and I'm back on the default shell and it is pretty fast. 

Anyway, i reckon it's a known issue with WSL. I have tried a few things from a few other relevant threads but hasn't really helped improve the terminal output speed.

**Comment by andresgongora** [comment]
Maybe you can disable all other features in the prompt, like git and python
support. They check for special files/env-variables that might be
problematic on Windows

**Comment by andresgongora** [comment]
Sure! Let's hope someone finds a solution to this eventually :)

**Comment by dannykurniawan80** [comment]
Hi @vinrepos, I had the same issue with WSL being slow processing the prompt. Then I found this:

https://yalneb.blogspot.com/2018/01/fancy-bash-promt.html

It seems that the script is much more simpler and doesn't add advanced features like git or python environment, but it suit my purpose. I'm guessing the script is an early version of the synth-shell-prompt.

The only drawback is, it is slightly more difficult to configure, since you need to read the script and find the exact variable to change, and no fancy git feature that I actually love. Good thing is there is no delay at all when displaying the prompt.

**Comment by andresgongora** [comment]
Hmmm if it's because of git or pyenvs, maybe you could try the new one and
disable all extra features from within the config

**Comment by dannykurniawan80** [comment]
Hi @andresgongora, first I disable all extra environments from config file as follows:

```bash
format="USER HOST PWD"
```

But it's still slow, so I try to disable the actual codes from `synth-shell-prompt.sh` as well:

```bash
...
prompt_command_hook()
{
        ## GET PARAMETERS
        ## This might be a bit redundant, but it makes it easier to maintain
        local elements=(${SSP_ELEMENTS[@]})
        local user=$USER
        local host=$HOSTNAME
        local path="$(shortenPath "$PWD" $SSP_MAX_PWD_CHAR $SSP_PWD_TRUNC_SYMBOL)" # bash-tools::shortenPath
        #local git_branch="$(getGitBranch)"
        #local pyenv="$(getPyenv)"
        #local kube="$(getKube)"
        #local tf="$(getTerraform)"
        local clock="$(date +"${SSP_CLOCK_FORMAT}")"
        ## ADAPT DYNAMICALLY ELEMENTS TO BE SHOWN
        ## Check if elements such as GIT and the Python environment should be
        ## shown and adapt the variables as needed. This usually implies removing
        ## the appropriate field from the "elements" array if the user set them
        #if [ -z "$git_branch" ]; then
        #       elements=( ${elements[@]/"GIT"} ) # Remove GIT from elements to be shown
        #fi
        #if [ -z "$pyenv" ]; then
        #       elements=( ${elements[@]/"PYENV"} ) # Remove PYENV from elements to be shown
        #fi
        #if [ -z "$tf" ]; then
        #       elements=( ${elements[@]/"TF"} ) # Remove TF from elements to be shown
        #fi
        #if [ -z "$kube" ]; then
        #       elements=( ${elements[@]/"KUBE"} ) # Remove KUBE from elements to be shown
        #fi
        ...
```

The delay slightly better, but still there...

**Comment by andresgongora** [comment]
I fear I don't have access to Windows to test this. Can you try to further
comment/isolate bits of the script. Once the culprit is identified we could
quickly fix it for everyone :)

**Comment by dannykurniawan80** [comment]
@andresgongora, I tried to modify `synt-shell-prompt.sh` and adding timestamp for execution of each line. Somewhat profiling execution time of each line in the script:

[profile.log](https://github.com/user-attachments/files/16493496/profile.log)

here's my modified script:

[synt-shell-prompt_debug.tar.gz](https://github.com/user-attachments/files/16493561/synt-shell-prompt_debug.tar.gz)

When I ran it, seems total time closing to 1.7 seconds (including profiling stuffs I think), but I think we can check how much time spent for each line of the codes from the log.

The configuration still same, I only change the following line in synth-shell-prompt.config:

```
format="USER HOST PWD"
```
